Test verification conflates error-line counts with structured failure counts

## Bug Description

The shepherd's test verification logs "N new error(s)" using two different counting methods that measure different things, producing confusing and contradictory output.

## Steps to Reproduce

Observed on shepherd run for #2436 (after Doctor attempt 1):

```
[03:13:04] [WARN] Baseline also fails but worktree introduces 166 new error(s)
[03:13:04] [ERROR] Tests failed (1 failed, 214 passed in 1.08s, 63s)
[03:13:04] [ERROR] Doctor introduced regressions (0 → 1 new errors), aborting test-fix loop
```

"166 new errors" vs "1 new error" — which is it?

## Root Cause

Two different counting methods both report "errors":

1. **Error-line counting** (`_extract_error_lines()`): Counts all output lines containing "fail", "error", etc. Used in the warning message at ~line 2298. Produced: **166**.

2. **Structured failure parsing** (`_compute_new_error_count()` → `_parse_failure_count()`): Parses test runner summary lines like "1 failed, 214 passed." Used in doctor regression detection. Produced: **1**.

The 166 error-indicator lines include stack traces, error context, and log lines that happen to contain "error" — not 166 failing tests.

## Proposed Fix

Use consistent terminology and prefer structured counts:
- When structured failure count is available, use it for BOTH the warning message and doctor regression detection
- Reserve error-line counting as a fallback only when parsing fails
- If error-line count is shown, label it clearly (e.g., "166 new error-indicator lines") to distinguish from "1 new test failure"

## Files

- `loom-tools/src/loom_tools/shepherd/phases/builder.py` — `_run_test_verification()` ~line 2298 and `_compute_new_error_count()` ~line 1699

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -570,7 +570,7 @@ def orchestrate(ctx: ShepherdContext) -> int:
                     # Doctor made things worse — abort immediately
                     log_error(
                         f"Doctor introduced regressions "
-                        f"({prev_error_count} → {current_error_count} new errors), "
+                        f"({prev_error_count} → {current_error_count} test failure(s)), "
                         f"aborting test-fix loop"
                     )
                     phase_durations["Builder"] = builder_total_elapsed

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -2393,10 +2393,21 @@ class BuilderPhase:
                                 ctx, primary_output
                             )
 
-                log_warning(
-                    f"Baseline also fails but worktree introduces "
-                    f"{len(new_errors)} new error(s)"
-                )
+                # Prefer structured failure count for clearer messaging.
+                # _compare_test_results returned False (structured
+                # comparison failed), but one side may still parse.
+                worktree_fc = self._parse_failure_count(worktree_output)
+                if worktree_fc is not None:
+                    log_warning(
+                        f"Baseline also fails but worktree has "
+                        f"{worktree_fc} test failure(s) "
+                        f"({len(new_errors)} error-indicator lines in diff)"
+                    )
+                else:
+                    log_warning(
+                        f"Baseline also fails but worktree introduces "
+                        f"{len(new_errors)} new error-indicator line(s)"
+                    )
 
         # Tests failed with new errors (or no baseline available)
         summary = self._parse_test_summary(primary_output)

